src/code/reducers/index.js

```diff
@@ -73,6 +75,33 @@ export default function reducer(state, action) {
         challengeProgress: progress
       });
     }
+    case actionTypes.EGG_SELECTED: {
+      return state.merge({
+        selectedEggIndex: action.eggIndex
+      });
+    }
+    case actionTypes.EGG_SUBMITTED: {
+      let progress = updateProgress(state, action.correct);
+
+      return state.merge({
+        submittedEggIndex: action.eggIndex,
+        submittedBasketIndex: action.basketIndex,
+        success: action.correct,
+        challengeProgress: progress
+      });
+    }
+    case actionTypes.EGG_REJECTED: {
+      return state.merge({
+        submittedEggIndex: null,
+        submittedBasketIndex: null
+      });
+    }
+    case actionTypes.EGG_ACCEPTED: {
+      return state.merge({
+        submittedEggIndex: null,
+        submittedBasketIndex: null
+      });
+    }
```

Comment:
It seems that this properties only exist so that we can hold onto a reference of which egg and basket were selected between the `submitEggForBasket` and the `accept/rejectEggFromBasket` actions, which have a modal dialog between them. The fact that these properties only exist between these times makes it seem like a work-around for the fact that we can't easily pass properties to the modal dialog.

It seems like updating the `MODAL_DIALOG_SHOWN` action to support arguments, and then the ModalDialogContainer could use those arguments when dispatching the action, would be a more generally-useful solution.